s2a_options.go

```diff
@@ -121,6 +125,34 @@ type ClientOptions struct {
 	// VerificationMode specifies the mode that S2A must use to verify the
 	// peer certificate chain.
 	VerificationMode VerificationModeType
+
+	// S2Av2 only.
+	// Optional fallback after dialing with S2Av2 fails.
+	FallbackOpts *FallbackOptions
+}
+
+// FallbackOptions contains the fallback options for S2Av2.
+type FallbackOptions struct {
+	// FallbackClientHandshakeFunc is used to specify fallback behavior when calling s2a.NewClientCreds().
+	// It will be called by s2av2TransportCreds's ClientHandshake function, after handshake with S2Av2 fails.
```

Comment:
s2av2TransportCreds is an internal implementation detail and should not be mentioned in a public-facing interface.